File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.15 Funkcja o prototypie void MM(double TAB[], int n, double &MX).cpp
#include <iostream>

double S1(int n) {
    if (n == 1) {
        return 1;
    } else {
        return 1.0 / n + S1(n - 1);
    }
}

int main() {
    int N;
    std::cout << "Podaj liczbê ca³kowit¹: ";
    std::cin >> N;
    std::cout << "Wynik: " << S1(N) << std::endl;
    return 0;
}

File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.15 Funkcja o prototypie void MM(double TAB[], int n, double &MX).py
def S1(n):
    if n == 1:
        return 1
    else:
        return 1 / n + S1(n - 1)

if __name__ == "__main__":
    N = int(input("Podaj liczbę całkowitą: "))
    print("Wynik:", S1(N))
